docs/APP-SETUP.md

```diff
@@ -0,0 +1,218 @@
+# RHDH Setup Guide For GitHub and GitLab
+
+## Table of Contents
+
+- [GitHub Personal Access Token (PAT)](#github-personal-access-token-pat)
+  - [Prerequisities](#prerequisites)
+  - [Procedure](#procedure)
+- [GitHub App](#github-app)
+  - [Procedure](#procedure-1)
+  - [App Permissions](#app-permissions)
+- [GitLab Personal Access Token (PAT)](#gitlab-personal-access-token-pat)
+  - [Prerequisites](#prerequisites-1)
+  - [Procedure](#procedure-2)
+- [GitLab App](#gitlab-app)
+  - [Procedure](#procedure-3)
+- [Quay Setup](#quay-setup)
+  - [Procedure](#procedure-4)
+- [Homepage / Callback / Webhook URLs](#homepage--callback--webhook-urls)
+  - [Homepage](#homepage-url)
+  - [Callback](#callback-url)
+  - [Webhook](#webhook-url)
+- [Troubleshooting](#troubleshooting)
+  - [Git Resource Not Accessible](#git-resource-not-accessible)
+  - [Pipeline Webhook Not Working](#pipeline-webhook-not-working)
+  - [Pipeline Failed To Push/Pull Image From Registry](#pipeline-failed-to-pushpull-image-from-registry)
+  - [Pipelines Fail To Run For Private Repositories](#pipelines-fail-to-run-for-private-repositories)
+
+## GitHub Personal Access Token (PAT)
+
+### Prerequisites
+
+- A GitHub account.
+
+### Procedure
```

Comment:
Might be better if we link directly the github docs for that? E.g https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-personal-access-token-classic

So we have less content to maintain

Reply:
Added the changes in https://github.com/redhat-ai-dev/ai-rhdh-installer/pull/11/commits/5ee65def7f3c28a3e81cede08d818b4b215e7d0f